### ISSUE TITLE
Improve arguments logging

### ISSUE THREAD

**Issue by pavlexander** [rosenbjerg/FFMpegCore#265]
Please make it easier to log argumets that are used for executables.

In my case I was using the snapshot command:

```
await FFMpeg.SnapshotAsync(
                    GetFileFullPath(request.InputFilePath),
                    GetFileFullPath(request.File.OriginalFilePath), 
                    null, 
                    TimeSpan.FromMilliseconds(1));
```

for some reason this command threw exception:

> Automatic encoder selection failed for output stream #0:0. Default encoder for format image2 (codec none) is probably disabled. Please choose an encoder manually. Error selecting an encoder for stream 0:0

the weird thing is - when I ran **presumably** the same command manually - it worked flawlessly:

`& ffmpeg.exe -ss 00:00:00.001 -i "C:\temp\files\uploaded\14102021\sfhaj2pt.1hi.mp4" -vframes 1 "C:\temp\files\uploaded\14102021\ssihmeer.p5k.png"`

so I started to wonder if the FFMpegCore library is doing something that it should not be doing and god it was hard to debug it! Without having any kind of logging implemented I could not check the end arguments that are used for `ffmpeg` executable; I ended up downloading the repository and including the project in my solution just to debug it. **A simple arguments logging feature would be very nice to have.**

On a side note, this is the command that library produces:
`& ffmpeg.exe -ss 00:00:00.001 -i "c:\temp\files\uploaded\14102021\nlvaabmc.yet.mp4" -map 0:0 -c:v png -vframes 1  "c:\temp\files\uploaded\14102021\0d1cz2su.1gz.png" -y`

for some reason this `-map 0:0` is appended even though I have not asked for it explicitly in any way. If this argument is removed form the command then snapshot is created nicely.  I am yet to figure out how to remove this argument.

So I guess another feature would be - maybe stop implicitly adding arguments that are not being asked for (see example above) or give us an option to opt-out? Is there a way to omit `-map 0:0`?

**Comment by pavlexander** [comment]
p.s. for now I ended up modifying the snapshot function and including the project in my solution..

```
        private static (FFMpegArguments, Action<FFMpegArgumentOptions> outputOptions) BuildSnapshotArguments(
            string input,
            IMediaAnalysis source,
            Size? size = null,
            TimeSpan? captureTime = null,
            int? streamIndex = null,
            int inputFileIndex = 0)
        {
            captureTime ??= TimeSpan.FromSeconds(source.Duration.TotalSeconds / 3);
            size = PrepareSnapshotSize(source, size);
            streamIndex = streamIndex == null ? 0 : source.VideoStreams.FirstOrDefault(videoStream => videoStream.Index == streamIndex).Index;

            return (FFMpegArguments
                .FromFileInput(input, false, options => options
                    .Seek(captureTime)),
                options => options
                    //.SelectStream((int)streamIndex, inputFileIndex) // <- this line
                    .WithVideoCodec(VideoCodec.Png)
                    .WithFrameOutputCount(1)
                    .Resize(size));
        }
```

**Comment by pavlexander** [comment]
actually nevermind my last solution. I think I found a better way:

```
var customArgs = FFMpegArguments
                    .FromFileInput(GetFileFullPath(request.InputFilePath), false, options =>
                    {
                        var captureTime = TimeSpan.FromSeconds(TimeSpan.FromMilliseconds(1).TotalSeconds / 3);
                        options.Seek(captureTime);
                    })
                    .OutputToFile(GetFileFullPath(request.File.OriginalFilePath), false, options =>
                    {
                        options.WithVideoCodec(VideoCodec.Png);
                        options.WithFrameOutputCount(1);
                    });

                var args = customArgs.Arguments; // debugging

                var success = await customArgs.ProcessAsynchronously();
```

hence bypassing snapshot command entirely

**Comment by rosenbjerg** [comment]
Looks like this was fixed in this PR #253 
Please reopen if that isn't the case
